### DISCUSSION TITLE
WEF Questions and Answers

### DISCUSSION THREAD

**Discussion by D3Ext** [D3Ext/WEF#5] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to ask about WEF and its usage.

If you want you can ask  about everything and I'll try to answer you as soon as possible. Don't be shy!

**Comment by Lion3323** [comment]
This may have been a better platform to ask. So when I am running WEF I choose to put my card in mon mode.  It then exits with my card in monitor mode but it wont let me launch unless it's in managed mode. What am I missing?

**Reply by D3Ext** [reply]
if I understand correctly you can't do the attacks unless the card is in managed mode right? I'm using a Parrot with an Alfa AWUS036ACM and it works always at the first, I run the program like this: 
```
wef -i wlan0
```

And once I entered the framework I select the 7 option and It starts properly my card in mon mode, and if I try to do any attack it works great. If you pass some images about the problem would help

**Reply by D3Ext** [reply]
What OS are you using?

**Reply by Lion3323** [reply]
Ubuntu. Alfa card.  I can run all the tools separately. I can manually take my card in and out of monitor mode. The issue is that when I use this tool it boots me out after the card is in mon mode. Then it wont let me back in unless its out of mon mode.

**Reply by D3Ext** [reply]
Ok, could you test it in Kali to see if it's a problem of the script or a problem in the Ubuntu support? I will test it in Ubuntu in depth and I tell you

Thanks for the help

**Reply by Lion3323** [reply]
Yea I'll give it a go in Kali and see if I get different results.  I'll post the results
Ty

**Reply by SODIVMOSDIMVOC65** [reply]
Why does  using WEF when appear "MAC address no found "

**Reply by D3Ext** [reply]
@qinzhoushishisanzhongxue could you add more info to your question?

**Reply by D3Ext** [reply]
Execute this command and post the output to help you:
```sh
ifconfig | grep "wlan0" | awk '{print $1}' | tr -d ':'
```
And change wlan0 to the name of your card interface

**Reply by fuzzyangel** [reply]
same thing for me
running wef in a early2008 macbook pro with Ubuntu 20.04.5 LTS

internal wifi gives
[X] MAC address not found 
after a "no such device" message that appears for a second

external usb wifi gives 
[X] Network interface not found
[X] MAC address not found 
after a "no such device" message that appears for a second

my wifi list is 
iw dev
phy#4
	Interface wlx00116bf0c181
		ifindex 13
		wdev 0x400000001
		addr 00:11:6b:f0:c1:81
		ssid fuznet2
		type managed
		channel 2 (2417 MHz), width: 20 MHz, center1: 2417 MHz
		txpower 20.00 dBm
phy#0
	Interface wls4
		ifindex 3
		wdev 0x1
		addr 00:22:41:f7:7f:00
		ssid SotNats
		type managed

**Comment by D3Ext** [comment]
@fuzzyangel which antenna are you using? If you manually try to start the monitor mode it gives errors?

**Comment by Alan-hub94** [comment]
I have a question, everything works correctly for me when I create an evil twin, the problem is when I try to look for the fake AP in my list of wireless networks, it doesn't appear to be able to connect, what can I be doing wrong?

**Comment by ErKbModifier** [comment]
Buenas, tengo una duda respecto a WEF con mi tarjeta wifi TP-Link TL-WN722N v2/v3 [Realtek RTL8188EUS] con el driver "8188eu
".

Cuando la pongo en modo monitor y sale en iwconfig Mode Monitor, WEF me dice que la sigo teniendo en mono Managed y no he podido probar la herramienta.
![image](https://user-images.githubusercontent.com/86917503/227716822-b357018b-30f6-4af4-bbc5-42b517fe45b7.png)

![image](https://user-images.githubusercontent.com/86917503/227716851-b0617d76-6498-4aa3-800f-643da19c51ef.png)

He probado varias cosas, pero no he conseguido nada que pudiera funcionar, no se si es porque estoy haciendo algo mal, pero los comandos que he usado son los siguientes:
`airmon-ng check kill`
`ifconfig wlan0 down`
`iw dev wlan0 set type monitor`
`ifconfig wlan0 up`

Otras veces cuando hago pruebas para intentar que funcione me muestra lo siguiente:
![image](https://user-images.githubusercontent.com/86917503/227717322-a445e170-d1bd-4638-8256-43fe2d79c7b7.png)

Sin embargo si he probado la tarjeta varias veces haciendo otras cosas y funciona:
![image](https://user-images.githubusercontent.com/86917503/227717416-8961c20e-df08-47b0-b8fa-ba645d6b6e3b.png)

Gracias por la ayuda.

**Reply by ErKbModifier** [reply]
Me he dado cuenta de algo raro y es que cada vez que uso la opción 7, cambia la MAC del dispositivo.
![image](https://user-images.githubusercontent.com/86917503/227718376-4d417f0b-bf5a-4d0a-81d0-f23d4b5e273e.png)
![image](https://user-images.githubusercontent.com/86917503/227718396-adeecec1-d081-4135-979f-b2601c3997e7.png)
![image](https://user-images.githubusercontent.com/86917503/227718408-d8da6ab4-13bd-4919-9a23-756119e6d7bf.png)
![image](https://user-images.githubusercontent.com/86917503/227718414-6273e2c8-53e9-44a5-8d50-2d547a34c9ab.png)
![image](https://user-images.githubusercontent.com/86917503/227718475-87759b4f-3472-42a9-89d5-d6c5aef09d03.png)

Después al hacer un reboot del sistema la MAC es la siguiente:
![image](https://user-images.githubusercontent.com/86917503/227718535-071d10c4-9553-4f2f-8860-d92281ec0783.png)
